actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -89,19 +96,28 @@ class ActionCable.Connection
 
     open: ->
       ActionCable.log("WebSocket onopen event")
-      @disconnected = false
-      @subscriptions.reload()
 
-    close: ->
+      if @isProtocolSupported()
+        ActionCable.log("WebSocket using '#{@getProtocol()}' subprotocol.")
+        @disconnected = false
+        @subscriptions.reload()
+      else
+        ActionCable.log("WebSocket got '#{unsupported_protocol}' as the chosen protocol. Stopping monitor and disconnecting.")
+        @disconnect(ActionCable.Events.Disconnected.ProtocolNotSupported)
```

Comment:
Ok yeah I see what you mean: if ActionCable were to support say SSEs or HTTP/2 in the future, reducing the surface area of WebSocket dependencies would be a good thing. :-)

In that case, we should just return an object with a `willAttemptReconnect` property set to `true/false`.